### PR TITLE
Support response validation

### PR DESCRIPTION
**Pull request by abenari**
This pull request contains 3 patches:
1. Add support for response validation.
With this patch it is easy to validate the response in unit tests with the documentation schema.
It help creating and maintaining a valid response documentation.

2. Supporting specification of array properties with non-string types

3. Allows creating a schema where objects are defined once and referenced by multiple methods.
When generating a client from the schema, the client code becomes easier to use.

### CODE REVIEW

**Comment by iNecas** [comment]
Looks promising. Would you mind updating the README to describe the new functionality?

File: README.rst [hunk 1]
```diff
@@ -842,6 +842,76 @@ The concern needs to be included to the controller after the methods are defined
...
+
+
+Example of the automated mechanism:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: ruby
+
+  require 'apipie/rspec/response_validation_helper'
+
+  RSpec.describe MyController, :type => :controller, :show_in_doc => true do
```
Comment (iNecas) [inline README.rst]: Missing indentation and final `end` for this block.

**Comment by abenari** [comment]
fixed response validation description to the readme file

**Comment by iNecas** [comment]
Apipie-rails-0.5.9 with this change has been pushed to Rubygems.

**Pull request merged**